### PR TITLE
Add the possibility to give an amount of experience with an item

### LINKED ISSUE #73: Add the possibility to give an amount of experience with an item
As a **User**
I would like to **set my items to give an amount of experience**
So that **I can implement Experience Candies from the last official games**

## Acceptance criterias
- In the Progression data block, add the `Experience increase` entry to manage an experience points value.

### PR DESCRIPTION
**Pull request by Otaku17**
Thank you for your contribution to the **Pokémon Studio** repo.

Before submitting this PR into the develop branch, please make sure:

- [x] Your code builds clean without any errors or warnings
- [x] You are following the [Code guidelines](../CodeGuidelines.md)
- [x] You tested your code to make sure it does what it is supposed to do

## Description

This update introduces support for creating items that can grant experience points in the Studio system. Here are the key changes:

- **Added "ExpGiveItem" category**: This new item category allows for assigning a quantity of experience when used. It is now available in the item creation and management options.
- **Updated types and validations**: Types and validations have been modified to include this new category. Items of type "ExpGiveItem" are now correctly identified and processed by the system.
- **Integrated into the item editor**: The item editor has been updated to include specific fields for "ExpGiveItem" items. Users can now specify the amount of experience granted directly in the interface.
- **Value management**: Experience values are validated to ensure they are reasonable and consistent with other item parameters.

This feature adds variety to the available items and introduces additional gameplay mechanics based on experience allocation.

## Tests Performed

The following tests were conducted to ensure the new functionality works as expected:

1. **Creation of New Items**: 
   - Created a new item of the type "ExpGiveItem".
   - Verified that the item was successfully created with the appropriate fields for experience points.
   - Confirmed that the JSON representation of the item includes `"klass": "ExpGiveItem"` and `"expCount"`.

2. **Modification of Existing Items**:
   - Modified an existing item to change its type to "ExpGiveItem".
   - Ensured that the item updated correctly and that the `"klass"` was updated to `"ExpGiveItem"`.
   - Checked that the `"expCount"` field was correctly saved and reflected in the updated JSON representation.

In both cases, the data was properly recorded, and the expected fields were present in the JSON files. This confirms that the new functionality is integrated correctly and that experience points are managed as intended.

### CODE REVIEW

File: src/views/components/database/item/ItemProgressData.tsx [hunk 1]
```diff
@@ -11,7 +11,12 @@ export const ItemProgressData = ({ dialogsRef }: ItemProgressDataProps) => {
   const { currentItem: item } = useItemPage();
   const { t } = useTranslation('database_items');
   const isItemEvBoost = item.klass === 'EVBoostItem';
+  const isItemLevelBoost = item.klass === 'LevelIncreaseItem';
+  const isItemExpBoost = item.klass === 'ExpGiveItem';
   const isDisabled = LOCKED_ITEM_EDITOR[item.klass].includes('progress');
+
+  console.log(isItemEvBoost, isItemExpBoost, isItemLevelBoost);
```
Comment (Palbolsky) [inline src/views/components/database/item/ItemProgressData.tsx]: Petit console.log qui a survécu :p

Comment (Otaku17) [reply src/views/components/database/item/ItemProgressData.tsx]: J'en oublie toujours un c'est pas possible x)

**Comment by Palbolsky** [comment]
J'ai trouvé un petit problème, il y a des données de l'ancienne classe d'objet qui restent dans le json.
Si tu passes de `Gain de niveau` à `Gain d'expérience` et inversement, tu trouveras dans le json `levelCount` et `expCount` alors qu'on ne doit avoir qu'un des deux.

Autre problème, mais qui était déjà là avant, mais si c'est facilement corrigeable regarde ce que tu peux faire 👀 

image
Quand on choisit `Gain d'EV`, la statistique ciblée n'est pas correct. Prend `Attaque` par défaut.

**Review by Palbolsky — APPROVED** [review]
C'est bon pour moi. J'approuve.

**Review by AerunDev — APPROVED** [review]
Pour moi c'est nickel, bien joué ! 😄

**Comment by AerunDev** [comment]
Je vais créer un ticket côté PSDK pour qu'on implémente la classe ExpGiveItem et qu'on ait une façon plus propre de gérer les choses que l'actuel système qui hardcode les valeurs. Bien joué encore une fois, je merge 👍

**Pull request merged**